Enable tvOS and watchOS (#147)
pytorch/cpuinfo: CMakeLists.txt

  ENDIF()
ENDMACRO()

# -- [ Determine whether building for Apple's desktop or mobile OSes

> SET(IS_APPLE_OS CMAKE_SYSTEM_NAME MATCHES "^(Darwin|iOS|tvOS|watchOS)$")

Reviewer: ```suggestion
IF(CMAKE_SYSTEM_NAME MATCHES "^(Darwin|iOS|tvOS|watchOS)$")
  SET(IS_APPLE_OS TRUE)
ELSE()
  SET(IS_APPLE_OS FALSE)
ENDIF()
```

Author: Thank you @malfet

# -- [ Determine target processor
SET(CPUINFO_TARGET_PROCESSOR "${CMAKE_SYSTEM_PROCESSOR}")

> IF(APPLE_SYSTEM)

Reviewer: One should not execute this path if `CMAKE_OSX_ARCHITECTURES` is empty.
```suggestion
IF(IS_APPLE_OS AND CMAKE_OSX_ARCHITECTURES MATCHES "^(x86_64|arm64.*)$")
```

  ENDIF()
ENDMACRO()

> -- [ Determine whether building for Apple's desktop or mobile OSes

Reviewer: ```suggestion
# -- [ Determine whether building for Apple's desktop or mobile OSes
```

SET(IS_APPLE_OS CMAKE_SYSTEM_NAME MATCHES "^(Darwin|iOS|tvOS|watchOS)$")

> 

Reviewer: ```suggestion
```

      "cpuinfo will compile, but cpuinfo_initialize() will always fail.")
    SET(CPUINFO_SUPPORTED_PLATFORM FALSE)
  ENDIF()
ELSEIF(NOT CPUINFO_TARGET_PROCESSOR MATCHES "^(i[3-6]86|AMD64|x86(_64)?|armv[5-8].*|aarch64|arm64.*|ARM64.*)$")
  MESSAGE(WARNING
    "Target processor architecture \"${CPUINFO_TARGET_PROCESSOR}\" is not supported in cpuinfo. "
    "cpuinfo will compile, but cpuinfo_initialize() will always fail.")
  SET(CPUINFO_SUPPORTED_PLATFORM FALSE)
ENDIF()

IF(NOT CMAKE_SYSTEM_NAME)
    MESSAGE(WARNING
      "Target operating system is not specified. "
      "cpuinfo will compile, but cpuinfo_initialize() will always fail.")
  SET(CPUINFO_SUPPORTED_PLATFORM FALSE)
ELSEIF(NOT CMAKE_SYSTEM_NAME MATCHES "^(Windows|WindowsStore|CYGWIN|MSYS|Darwin|Linux|Android)$")

>   IF(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.14" AND NOT APPLE_SYSTEM)

Reviewer: ```suggestion
  IF(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.14" AND NOT IS_APPLE_OS)
```

      LIST(APPEND CPUINFO_SRCS
        src/x86/linux/init.c
        src/x86/linux/cpuinfo.c)

>     ELSEIF(APPLE_SYSTEM)

Reviewer: ```suggestion
    ELSEIF(IS_APPLE_OS)
```

      ELSEIF(CMAKE_SYSTEM_PROCESSOR MATCHES "^(aarch64|arm64)$")
        LIST(APPEND CPUINFO_SRCS src/arm/linux/aarch64-isa.c)
      ENDIF()

>     ELSEIF(IOS OR

Reviewer 2: would it be possible to rename `IOS` to `APPLE_MOBILE_OS` and then set `APPLE_OS` = (iOS or tvOS or watchOS), and avoid having to check the specific strings in multiple places?

Author: Regular expression matching came into my mind. Does it make sense?

Reviewer 2: i was mostly trying to find a way to put the (ios or watchos or tvos) logic in one place and store that into a variable and then everywhere else just check the variable.

That's unless, of course, you wanted different behavior for the different os's somewhere...

Author: Thank you for the clarification @wconstab . Done. Please take another look. Thanks!

Reviewer 2: looks better to me! thanks.

Let me know if CI has issues that aren't your fault- gh actions has been unstable recently.  If its passing CI I think its good to land.  cc @malfet

      ELSEIF(CMAKE_SYSTEM_PROCESSOR MATCHES "^(aarch64|arm64)$")
        LIST(APPEND CPUINFO_SRCS src/arm/linux/aarch64-isa.c)
      ENDIF()

>     ELSEIF(APPLE_SYSTEM AND CPUINFO_TARGET_PROCESSOR MATCHES "arm64.*")

Reviewer: ```suggestion
    ELSEIF(IS_APPLE_OS AND CPUINFO_TARGET_PROCESSOR MATCHES "arm64.*")
```

      src/linux/multiline.c
      src/linux/cpulist.c
      src/linux/processors.c)

>   ELSEIF(APPLE_SYSTEM)

Reviewer: ```suggestion
  ELSEIF(IS_APPLE_OS)
```
